vllm/attention/ops/triton_flash_attention.py

```diff
@@ -335,487 +668,736 @@ def attn_fwd(
+                if seqlen_k < BLOCK_N:
+                    n_extra_tokens = BLOCK_N - seqlen_k
+                elif seqlen_k % BLOCK_N:
+                    n_extra_tokens = seqlen_k % BLOCK_N
+                USE_PADDED_HEAD: tl.constexpr = (IS_ACTUAL_BLOCK_DMODEL
+                                                 != BLOCK_DMODEL)
+
+                # Compute pointers for all the tensors used in this kernel.
+                q_offset = (Q + off_z * stride_qz + off_h_q * stride_qh +
+                            cu_seqlens_q_start * stride_qm)
+                q_ptrs = (q_offset + offs_m[:, None] * stride_qm +
+                          offs_d[None, :] * stride_qk)
+                k_offset = (K + off_z * stride_kz + off_h_k * stride_kh +
+                            cu_seqlens_k_start * stride_kn)
+                k_ptrs = (k_offset + offs_d[:, None] * stride_kk +
+                          offs_n[None, :] * stride_kn)
+                v_offset = (V + off_z * stride_vz + off_h_k * stride_vh +
+                            cu_seqlens_k_start * stride_vk)
+                v_ptrs = (v_offset + offs_n[:, None] * stride_vk +
+                          offs_d[None, :] * stride_vn)
+                # Compute pointers for all scale tensors used in this kernel.
+
+                IS_EIGHT_BIT_GEMM: tl.constexpr = IS_EIGHT_BIT & (
+                    not IS_EIGHT_BIT_KV)
+                if IS_EIGHT_BIT:
+                    if k_descale_has_singleton:
+                        k_descale_ptrs = k_descale_ptr
+                    else:
+                        k_descale_ptrs = k_descale_ptr + off_h_k
+
+                    if v_descale_has_singleton:
+                        v_descale_ptrs = v_descale_ptr
+                    else:
+                        v_descale_ptrs = v_descale_ptr + off_h_k
+
+                    if not IS_EIGHT_BIT_KV:
+                        if q_descale_has_singleton:
+                            q_descale_ptrs = q_descale_ptr
+                        else:
+                            q_descale_ptrs = q_descale_ptr + off_h_q
+                    if USE_P_SCALE:
+                        if p_descale_has_singleton:
+                            p_scale_ptrs = p_scale_ptr
+                            p_descale_ptrs = p_descale_ptr
+                        else:
+                            p_scale_ptrs = p_scale_ptr + off_h_q
+                            p_descale_ptrs = p_descale_ptr + off_h_q
+
+                if USE_BIAS:
+                    # Note: might get large enough to overflow on some configs
+                    bias_offset = off_h_q * stride_bh
+                    bias_ptrs = (bias + bias_offset +
+                                 offs_m[:, None] * stride_bm +
+                                 offs_n[None, :] * stride_bn)
+                else:
+                    bias_ptrs = None
+
+                if USE_ALIBI:
+                    a_offset = off_z * stride_az + off_h_q * stride_ah
```

Comment:
similar, does it make sense to do:
```suggestion
                    a_offset = off_z * tl.cast(stride_az, tl.int64) + off_h_q * tl.cast(stride_ah, tl.int64)
```

Reply:
Done